tutorials/HDF5/Subset-HDF5-hyperspectral.Rmd

```diff
@@ -0,0 +1,322 @@
+
+
+```{r plot-example-band}
+# Extract or "slice" data for band 58 from the HDF5 file
+b58 <- h5read(f,name = "SJER/Reflectance/Reflectance_Data",
+             index=list(58,NULL,NULL))
+h5closeAll()
+
+# convert from array to matrix
+b58 <- b58[1,,]
+
+# Make a plot to view this band
+image(log(b58), col=grey(0:100/100))
+
+```
+
+As we can see here, this hyperspectral reflectance tile contains a school campus that is under construction. There are many different land cover types contained here, which makes it a great example! Perhaps the most unique feature shown is in the bottom right corner of this image, where we can see the tip of a small reservoir. Let's be sure to capture this feature in our spatial subset, as well as a few other land cover types (irrigated grass, trees, bare soil, and buildings).
+
+While raster images count their pixels from the top left corner, we are working with a matrix, which counts its pixels from the bottom left corner. Therefore, rows are counted from the bottom to the top, and columns are counted from the left to the right. If we want to sample the bottom right quadrant of this image, we need to select rows 1 through 500 (bottom half), and columns 501 through 1000 (right half). Note that, as above, the `index=` argument in `h5read()` requires a `list` of three dimenstions for this example - in the order of bands, columns, rows. 
+
+```{r plot-example-band-subset}
+subset_rows <- 1:500
+subset_columns <- 501:1000
+# Extract or "slice" data for band 44 from the HDF5 file
+b58 <- h5read(f,name = "SJER/Reflectance/Reflectance_Data",
+             index=list(58,subset_columns,subset_rows))
+h5closeAll()
+
+# convert from array to matrix
+b58 <- b58[1,,]
+
+# Make a plot to view this band
+image(log(b58), col=grey(0:100/100))
+
+```
+
+Perfect - now we have a spatial subset that includes all of the different land cover types that we are interested in investigating. 
+
+<div id="ds-challenge" markdown="1">
+### Challenge: Pick your subset
+
+1. Pick your own area of interest for this spatial subset, and find the rows and columns that capture that area. Can you include some solar panels, as well as the water body?
+
+2. Does it make a difference if you use a band from another part of the electromagnetic spectrum, such as the near-infrared? Hint: you can use the 'wavelengths' function above while removing the `index=` argument to get the full list of band wavelengths.
+
+</div>
+
+## Extracting the example sbset
```

Comment:
typo

Reply:
Also, is "example subset" the most useful header? Could it be just be "Extracting a subset"?

Reply:
Yes this is a good suggestion, and has been incorporated in the latest version.